x/mongo/driver/operation.go

```diff
@@ -1827,7 +1828,7 @@ func (Operation) decodeOpReply(wm []byte) opReply {
 	return reply
 }
 
-func (op Operation) decodeResult(opcode wiremessage.OpCode, wm []byte) (bsoncore.Document, error) {
+func (op Operation) decodeResult(opcode wiremessage.OpCode, wm []byte, isCSOT bool) (bsoncore.Document, error) {
```

Comment:
Suggest not use a bool here and instead pass in the context and use `csot.IsTimeoutContext(ctx)` within `decodeResult`. This ensures that the source of CSOT-ness is derived from a context.

Reply:
Sounds good, will do.

Reply:
Updated to pass the ctx.